includes/User.php

```diff
@@ -4617,4 +4617,22 @@ public function jsonSerialize() {
 			'mName' => $this->mName,
 		];
 	}
+
+	public static function getIndexedUserNames( $userIds ): array {
+		global $wgExternalSharedDB;
+		if ( count( $userIds ) === 0 ) {
```

Comment:
You can port this check to `User::whoAre()`, although using `count()` for `0`-checks is subopimal. In most cases this array will not be empty. `$ids == []` will be much faster than `count($ids) == 0`, as internally the first comparison resolves after reaching the first element of the array, whereas `count()` requires iterating over the entire table.  Even `count($ids)` on an empty array is slower than `$ids == []` because of internal counter initialization.  It's circa 2 ms per execution.  Given our scale, the frequency of using this method in the code and possible similar checks before calling this method (e.g. `if ($ids != []) { $names = User::whoAre($ids); }`), it's worth to optimize such cases.